tests/unit/authenticators/devise-test.js

```diff
@@ -50,46 +49,47 @@ describe('DeviseAuthenticator', () => {
       server.post('/users/sign_in', () => [201, { 'Content-Type': 'application/json' }, '{ "token": "secret token!", "email": "email@address.com" }']);
     });
 
-    it('sends an AJAX request to the sign in endpoint', function() {
-      return authenticator.authenticate('identification', 'password').then(() => {
-        let [request] = server.handledRequests;
+    it('sends an AJAX request to the sign in endpoint', async function() {
+      await authenticator.authenticate('identification', 'password');
+      let [request] = server.handledRequests;
 
-        expect(request.url).to.eql('/users/sign_in');
-        expect(request.method).to.eql('POST');
-        expect(JSON.parse(request.requestBody)).to.eql({ user: { email: 'identification', password: 'password' } });
-        expect(request.requestHeaders['content-type']).to.eql('application/json');
-        expect(request.requestHeaders.accept).to.eql('application/json');
-      });
+      expect(request.url).to.eql('/users/sign_in');
+      expect(request.method).to.eql('POST');
+      expect(JSON.parse(request.requestBody)).to.eql({ user: { email: 'identification', password: 'password' } });
+      expect(request.requestHeaders['content-type']).to.eql('application/json');
+      expect(request.requestHeaders.accept).to.eql('application/json');
     });
 
     describe('when the authentication request is successful', function() {
       beforeEach(function() {
         server.post('/users/sign_in', () => [201, { 'Content-Type': 'application/json' }, '{ "token": "secret token!", "email": "email@address.com" }']);
       });
 
-      it('resolves with the correct data', function(done) {
-        authenticator.authenticate('email@address.com', 'password').then((data) => {
-          expect(true).to.be.true;
-          expect(data).to.eql({ token: 'secret token!', email: 'email@address.com' });
-          done();
-        });
+      it('resolves with the correct data', async function() {
+        let data = await authenticator.authenticate('email@address.com', 'password');
+
+        expect(data).to.eql({ token: 'secret token!', email: 'email@address.com' });
       });
 
       describe('when the server returns incomplete data', function() {
-        it('fails when token is missing', function() {
+        it('fails when token is missing', async function() {
           server.post('/users/sign_in', () => [201, { 'Content-Type': 'application/json' }, '{ "email": "email@address.com" }']);
 
-          return authenticator.authenticate('email@address.com', 'password').catch((error) => {
+          try {
+            await authenticator.authenticate('email@address.com', 'password');
```

Comment:
Might be good to do something like https://github.com/simplabs/ember-simple-auth/pull/2013/files#diff-bb12f19912e2ea2cf2254d67032dbc05R27, if `authenticate` breaks and stops failing with an incorrect response, the test would still pass.

Reply:
yes, good point - fixed 👍

Reply:
That pattern is in several places 😅there should be a nicer way to achieve this.

Reply:
I fixed the others as well - cannot think of a better solution for now 🤔